inst/shiny/DiagnosticsExplorer/global.R

```diff
@@ -1,59 +1,92 @@
 library(magrittr)
 
+source("R/Init.R")
 source("R/Tables.R")
 source("R/Other.R")
 source("R/Plots.R")
 source("R/Results.R")
 
+connectionPool <- NULL
+defaultLocalDataFolder <- "data"
+defaultLocalDataFile <- "PreMerged.RData"
+
 if (!exists("shinySettings")) {
-  if (file.exists("data")) {
-    shinySettings <- list(dataFolder = "data")
-  } else {
-    shinySettings <- list(dataFolder = "S:/examplePackageOutput")
+  shinySettings <- list(
+    connectionDetails = getDbConnectionDetails(),
+    resultsDatabaseSchema = getResultsDatabaseSchema(),
+    cdmDatabaseSchema = getCdmDatabaseSchema(),
+    dataFolder = defaultLocalDataFolder
+  )
+} else {
+  if (is.null(shinySettings$connectionDetails)) {
+    shinySettings$connectionDetails <- getDbConnectionDetails()
+  }
+  if (is.null(shinySettings$dataFolder)) {
+    shinySettings$dataFolder <- defaultLocalDatFolder
+  }
+  if (is.null(shinySettings$resultsDatabaseSchema)) {
+    shinySettings$resultsDatabaseSchema <- getResultsDatabaseSchema()
+  }
+  if (is.null(shinySettings$cdmDatabaseSchema)) {
+    shinySettings$cdmDatabaseSchema <- getCdmDatabaseSchema()
   }
 }
-dataFolder <- shinySettings$dataFolder
-
-suppressWarnings(
-  rm(
-    "analysisRef",
-    "temporalAnalysisRef",
-    "temporalTimeRef",
-    "covariateValue",
-    "covariateRef",
-    "temporalCovariateValue",
-    "temporalCovariateRef",
-    "concept",
-    "conceptSynonym",
-    "vocabulary",
-    "domain",
-    "relationship",
-    "conceptAncestor",
-    "conceptRelationship",
-    "cohort",
-    "cohortCount",
-    "cohortOverlap",
-    "conceptSets",
-    "database",
-    "incidenceRate",
-    "includedSourceConcept",
-    "inclusionRuleStats",
-    "indexEventBreakdown",
-    "orphanConcept",
-    "timeDistribution",
-    "visitContext"
-  )
-)
 
-if (file.exists(file.path(dataFolder, "PreMerged.RData"))) {
-  writeLines(paste0("Using merged data detected in folder '", dataFolder, "'"))
-  load(file.path(dataFolder, "PreMerged.RData"))
+
+
+suppressWarnings(rm(list = resultsGlobalReferenceTables))
+suppressWarnings(rm(list = cdmGlobalReferenceTables))
+
+
+# Cleanup the database connPool if it was created
+# (borrowed from https://github.com/ohdsi-studies/Covid19CharacterizationCharybdis/blob/master/inst/shiny/CharybdisResultsExplorer/global.R)
+onStop(function() {
+  if (!is.null(connectionPool)) {
+    if (DBI::dbIsValid(connectionPool)) {
+      writeLines("Closing database pool")
+      pool::poolClose(connectionPool)
+    }
+  }
+})
+
+
+if (is.null(shinySettings$connectionDetails)) {
+  warning("No database connection details. Looking for local data.")
+  if (is.null(shinySettings$dataFolder)) {
+    stop("No mechanism to load data.")
+  } else {
```

Comment:
Else logic after stop is redundant so this could be tidied up

Reply:
thanks @azimov code cleaned up. can you please review again?